### PR TITLE
Mainly a bunch of changes around the specialised classes of `tern:Value`

### PR DESCRIPTION
**Pull request by edmondchuc**
Closes https://github.com/ternaustralia/ontology_tern/issues/26

Mainly a bunch of changes around the specialised classes of `tern:Value`. Removal of `tern:Instrument` in favour of tern:System (aligned to `ssn:System`). More changes were done, see changelog or the diff.

See changelog:

## Attribute
Update the `skos:definition` of the class.

### `dcterms:type`
All classes may have `dcterms:type`. There should be no cardinality here.

Removing `sh:maxCount = 1`. Add `sh:description`.

### `void:inDataset`
Remove `sh:minCount = 1` and `sh:maxCount = 1`

Add description.

### `tern:attribute`
Add description.

### `tern:hasSimpleValue`
Add description.

### `tern:hasValue`
Add description

### `tern:isAttributeOf`
Add description. Remove cardinality.

## `tern:Count`
- Class removed in favour of `tern:Integer`

## `tern:Integer`
- Create class
- Add class as a subclass of `tern:Value`

## `tern:Float`
- Create class
- Add class as a subclass of `tern:Value`

## `tern:QuantitativeRange`
- Class removed since use cases for requiring this is unlikely

## `tern:Percent`
- Class removed in favour of using `tern:Float` with the correct `tern:unit` from QUDT.

## `tern:PercentRange`
- Class removed due to unlikely use 

## `tern:QuantitativeMeasure`
- Class removed in favour of `tern:Float`

## `tern:Plot`
- This class has been removed

## `tern:System`
- Add new property called `tern:systemType`

## `tern:Instrument`
- Class removed

## `tern:Sampler`
- SHACL constraints added

## `tern:System`
- Added property `tern:systemType` and `dcterms:type`

## `tern:Concept`
- Class removed in favour of `tern:IRI`

## `tern:Observation`
- `sosa:hasResult` `sh:class` is an `AND` between `tern:Value` and `tern:Result`.

## `tern:Result`
- No longer a subclass of `tern:Value`

## Diagram overview

See the new class hierarchy of `tern:Value` and its subclasses.

class-overview

### CODE REVIEW

**Review by dr-shorthair — APPROVED** [review]

**Comment by dr-shorthair** [comment]
Are negative counts possible? Should it be `positiveInteger` ?

**Comment by dr-shorthair** [comment]
Should `tern:IRI` really be `tern:Thing` ?

**Comment by edmondchuc** [comment]
> Are negative counts possible? Should it be positiveInteger ?

@dr-shorthair yes I think `Integer` includes negative values. The value may not always be a result of an observation. Example, a negative integer value may be part of an attribute of a sampler like the configuration value for an electronic device. 

> Should tern:IRI really be tern:Thing ?

I think `tern:IRI` is more intuitive here. I understand why you have suggested `tern:Thing` but I'd like to avoid having to explain to non-OWL people what a `Thing` is (a resource identified by an IRI).

If everything else is all good @dr-shorthair, I'll merge this by midday. Cheers.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,15 +18,44 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- tern:Observation
+  - sosa:hasResult sh:class is an AND between tern:Value and tern:Result.
+- tern:Result no longer a subclass of tern:Value
 
 ### Removed
+
 - Import of datatype vocabulary
 - tern:isAttributeOf a sub-property of ssn:isPropertyOf because the TopBraid Composer diagram view is inferring that the range of values is sosa:FeatureOfInterest.
 - Some OWL restrictions as it doubles up in the diagram view in TopBraid Composer with the SHACL constraints.
 - Slowly remove OWL restrictions as they are unused besides expressing the OWL side of the model. More in favour of SHACL now.
```
Comment (dr-shorthair) [inline CHANGELOG.md:48]: Good.

**Review by dr-shorthair — APPROVED** [review]

**Pull request merged**